### ISSUE TITLE
Erweiterung für Xtra?

### ISSUE THREAD

**Issue by musicmasterowl** [rg-engineering/ioBroker.amtronwallbox#1]
Hi René, 

ich bin gerade dabei, Deinen Adapter für meine Amtron Xtra 22 C2 zu testen. Zum einen bekomme ich einen Log-Eintrag, dass der Typ "Compact" noch nicht implementiert ist:

<html><body>

  | system type Compact string not yet implemented
-- | --

</body>
</html>

Hier kann ich (aktuell) noch keinen anderen Typ auswählen. Deshalb zum anderen eine Frage: Hast Du vor, auch die Xtra zu implementieren? Das fänd ich sehr hilfreich und würd das gerne mit Dir zusammen testen (auch wenn ich eher ein DAU bin ;-)

**Versions:**  
 - Adapter version: 0.0.3
 - JS-Controller version: 4.0.23
 - Node version: 16.18.1

Viele Grüße, 

musicmasterowl

**Comment by rg-engineering** [comment]
Du müsstest doch wenigstens den "ChargeControl" auswählen können:

![grafik](https://user-images.githubusercontent.com/8238668/206895722-0e5764e2-f111-4a56-a2d9-b5c5e6e07162.png)


Die anderen würde ich gern implementieren, da ich aber keinen Zugriff auf solche Boxen habe, ist das schwierig. Ich würde also gern die Unterstützung annehmen... Ich muss nur schauen, ob mir auch die entsprechende Doku zur Verfügung steht.

**Comment by musicmasterowl** [comment]
Hi René, 

ha, jetzt hab ich die Xtra gefunden - das Scroll down Menü war sehr verdeckt. Gibt es eine Möglichkeit, das offensichtlicher zu gestalten?

![grafik](https://user-images.githubusercontent.com/120299017/206901976-ab29d7f4-ef12-4c56-8dd7-59fb4c9d1929.png)

Aber Daten kommen weder unter Compact, noch Charge Control noch Xtra rein. 

Soweit ich das überblicke, ist lephistos Dokumentation (https://github.com/lephisto/amtron) für die Premiums, aber die Xtra hat wohl weitestgehend die gleichen Schnittstellen. Zumindest kann ich sie per JS auslesen, hab aber noch nicht alle Variablen getestet. Insofern passen meine Settings (IP-Adresse, API-Schlüssel).

Viele Grüße, 

musicmasterowl

**Comment by rg-engineering** [comment]
Den lephistos-Fork von https://github.com/orlopau/amtron kannte ich noch nicht. Okay ich, schau mal. ob ich das in den nächsten Tagen mal ansatzweise umsetzen kann...

**Comment by rg-engineering** [comment]
@musicmasterowl Ich habe mal mit der Implementierung begonnen. Im Moment geht noch nicht viel, ich bräuchte aber deine Hilfe für den ersten Test: Beim Start des Adapters werden jetzt andere Datenpunkte angelegt, nämlich die, die das Interface für die Xtra zur Verfügung stellt. Idealerweie löschst du deinen aktuellen Datenbaum für den Adapter, nachdem du die Version installiert hast. Dann wird ein get-Kommando an die Box geschickt. Die Daten, die die Box schickt, logge ich nur im debug log aus, brauche sie aber für den nächsten step der Implementierung. Kannst du mir dieses log zukomen lassen?

**Comment by rg-engineering** [comment]
Lesen von
* Info
* Charge Data
* Statistik-Daten
sollte jetzt funktionieren.
Schreiben folgt noch...

**Comment by musicmasterowl** [comment]
Funktioniert, die Liste der warnungen ist kürzer geworden:

<html><body>


2022-12-17 14:18:00.404  - warn:  amtronwallbox.0 (4942) State "amtronwallbox.0.AMTRON.charge.TMaxCurrT1"  has no existing object, this might lead to an error in future versions
2022-12-17 14:18:00.456  - warn:  amtronwallbox.0 (4942) State "amtronwallbox.0.AMTRON.charge.TBeginH_T1"  has no existing object, this might lead to an error in future versions
2022-12-17 14:18:00.462  - warn:  amtronwallbox.0 (4942) State "amtronwallbox.0.AMTRON.charge.TBeginM_T1"  has no existing object, this might lead to an error in future versions
2022-12-17 14:18:00.467  - warn:  amtronwallbox.0 (4942) State "amtronwallbox.0.AMTRON.charge.TPriceT1"  has no existing object, this might lead to an error in future versions
2022-12-17 14:18:00.473  - warn:  amtronwallbox.0 (4942) State "amtronwallbox.0.AMTRON.charge.TMaxCurrT2"  has no existing object, this might lead to an error in future versions
2022-12-17 14:18:00.478  - warn:  amtronwallbox.0 (4942) State "amtronwallbox.0.AMTRON.charge.TBeginH_T2"  has no existing object, this might lead to an error in future versions
2022-12-17 14:18:00.485  - warn:  amtronwallbox.0 (4942) State "amtronwallbox.0.AMTRON.charge.TBeginM_T2"  has no existing object, this might lead to an error in future versions
2022-12-17 14:18:00.499  - warn:  amtronwallbox.0 (4942) State "amtronwallbox.0.AMTRON.charge.TPriceT2"  has no existing object, this might lead to an error in future versions
2022-12-17 14:18:00.505  - warn:  amtronwallbox.0 (4942) State  "amtronwallbox.0.AMTRON.charge.TRemoteCurr" has no existing object, this  might lead to an error in future versions
2022-12-17 14:18:00.556  - warn:  amtronwallbox.0 (4942) State  "amtronwallbox.0.AMTRON.charge.TSolarPrice" has no existing object, this  might lead to an error in future versions
2022-12-17 14:18:00.562  - warn:  amtronwallbox.0 (4942) State "amtronwallbox.0.AMTRON.charge.TExcessNrg"  has no existing object, this might lead to an error in future versions
2022-12-17 14:18:00.838  - info:  amtronwallbox.0 (4942) State value to set for  "amtronwallbox.0.AMTRON.Statistics.Annual.Years" has to be type "string"  but received type "object"


</body>
</html>


die Statistikwerte kommen alle rein, nur bei Year sieht es seltsam aus:

![grafik](https://user-images.githubusercontent.com/120299017/208243976-e171b694-c677-4276-87df-5246d0810d2b.png)

**Comment by rg-engineering** [comment]
> die Statistikwerte kommen alle rein, nur bei Year sieht es seltsam aus:

nein, das ist so gewollt. Hier kommt ein Json-Objekt mit den ganzen Daten pro Jahr. Das kann ich nicht auf einzelne Datenpunkte aufteilen, wären einfach zu viele DP und außerdem dynamisch, da jedes Jahr neue datenpunkte hinzukommen. Hier muss der user dann die Aufteilung machen, wie er es benötigt...

**Comment by rg-engineering** [comment]
@musicmasterowl wenn du möchtest, kannst du auch mal probieren, ob ds Schreiben von Werten jetzt funktioniert. Derzeit sollten nur die Werte von [hier](https://github.com/lephisto/amtron/blob/master/docs/api/DevInfo/post.md) funktionieren

**Comment by musicmasterowl** [comment]
Schreiben bei Battery geht, bei allen anderen Einträgen bekomme ich eine Fehlermeldung:

2022-12-21 10:45:01.930 - error: amtronwallbox.0 (9329) exception in write_MHCP_DevInfo [AxiosError: Request failed with status code 400]
